app/transport/rest/user_test.go

```diff
@@ -0,0 +1,149 @@
+	}{
+		"success": {
+			id:                 "a6703a6b-c054-4ebd-b782-2a96bd4f771a",
+			expectedStatusCode: http.StatusOK,
+			expectedResponseBody: `{ 
+				"id": "a6703a6b-c054-4ebd-b782-2a96bd4f771a", 
+				"first_name": "Jon", 
+				"last_name":  "Doe", 
+				"email":    "john@emasadfasilsdf.com", 
+				"created_at": "2022-11-01T16:31:40.400825Z"	
+			}`,
+			getByIdFunc: func(ctrl *gomock.Controller, id string) *MockUserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+				mock.EXPECT().
+					GetByID(gomock.Eq(id)).
+					Return(&entities.User{
+						ID:        "a6703a6b-c054-4ebd-b782-2a96bd4f771a",
+						FirstName: "Jon",
+						LastName:  "Doe",
+						Email:     "john@emasadfasilsdf.com",
+						CreatedAt: "2022-11-01T16:31:40.400825Z"}, nil).
+					Times(1)
+
+				return mock
+			},
+		},
+		"invalid_short_id": {
+			id:                 "1234",
+			expectedStatusCode: http.StatusBadRequest,
+			expectedResponseBody: `{
+				"message": "Bad request", 
+				"details": "invalid UUID length: 4"
+			}`,
+			getByIdFunc: func(ctrl *gomock.Controller, id string) *MockUserUsecase {
+				return nil
+			},
+		},
+		"invalid_long_id": {
+			id:                 "a6703a6b-c054-4ebd-b782-2a96bd4fasfd771asdfadfsfadsaa6703a6b-c054-4ebd-b782-2a96bd4f771aa6703a6b-c054-4ebd-b782-2a96bd4f771aa670",
+			expectedStatusCode: http.StatusBadRequest,
+			expectedResponseBody: `{
+				"message": "Bad request", 
+				"details": "invalid UUID length: 128"
+			}`,
+			getByIdFunc: func(ctrl *gomock.Controller, id string) *MockUserUsecase {
+				return nil
+			},
+		},
+		"zero_id": {
+			id:                 "",
+			expectedStatusCode: http.StatusBadRequest,
+			expectedResponseBody: `{ 
+				"message": "Bad request", 
+				"details": "invalid UUID length: 0" 
+			}`,
+			getByIdFunc: func(ctrl *gomock.Controller, id string) *MockUserUsecase {
+				return nil
+			},
+		},
+		"wrong_id": {
+			id:                 "a6703a6b-c054-4ebd-b782-2a96bd4f771a",
+			expectedStatusCode: http.StatusNotFound,
+			expectedResponseBody: `{
+				"message": "Not found",
+				"details": "no user found in DB"
+			} `,
+			getByIdFunc: func(ctrl *gomock.Controller, id string) *MockUserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+				mock.EXPECT().
+					GetByID(gomock.Eq(id)).
+					Return(nil, globals.ErrNotFound)
+
+				return mock
+			},
+		},
+		"unexpected_error": {
+			id:                 "a6703a6b-c054-4ebd-b782-2a96bd4f771a",
+			expectedStatusCode: http.StatusInternalServerError,
+			expectedResponseBody: `{ 
+				"message": "Internal server error", 
+				"details": ""
+			}`,
+			getByIdFunc: func(ctrl *gomock.Controller, id string) *MockUserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+				mock.EXPECT().
+					GetByID(gomock.Eq(id)).
+					Return(nil, errors.New("unexpected error"))
+
+				return mock
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
```

Comment:
Please think about a defer inside a loop. It may cause problems.

Reply:
@vpakhuchyi As far as I understand ```defers``` live in the scope or `func` and it is safe inside each iteration. but I could be wrong
```golang
for {
     func() { defer foo() }()
 }
```